adminapi/cli.py

```diff
@@ -0,0 +1,131 @@
+"""igserver - The command line interface
+
+Copyright (c) 2017, InnoGames GmbH
+"""
+
+from __future__ import print_function
+
+from argparse import ArgumentParser, ArgumentTypeError
+from distutils.util import strtobool
+
+from adminapi import _api_settings
+from adminapi.dataset import QuerySet
+from adminapi.dataset.base import MultiAttr
+from adminapi.utils.parse import parse_query
+
+
+def parse_args():
+    parser = ArgumentParser('igserver')
+    parser.add_argument('quals', nargs='+')
+    parser.add_argument(
+        '-1',
+        '--one',
+        action='store_true',
+        help='Make sure exactly one server matches with the query',
+    )
+    parser.add_argument(
+        '-a',
+        '--attrs',
+        nargs='+',
+        default=['hostname'],
+        help='The attributes to fetch',
+    )
+    parser.add_argument(
+        '-o',
+        '--orderby',
+        nargs='+',
+        default=['hostname'],
+        help='Attributes to order by the result',
+    )
+    parser.add_argument(
+        '-r',
+        '--reset',
+        nargs='+',
+        default=[],
+        help='Reset multi attributes',
+    )
+    parser.add_argument(
+        '-u',
+        '--update',
+        nargs='+',
+        type=attr_value,
+        default=[],
+        help='The attributes to update',
+    )
+
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+    servers = (
+        QuerySet(
+            # TODO: Avoid .join()
+            filters=parse_query(' '.join(args.quals)),
+            auth_token=_api_settings['auth_token'],
+            timeout=_api_settings['timeout_dataset'],
+        )
+        .restrict(*(args.attrs + args.reset + [k for k, v in args.update]))
+        .order_by(*args.orderby)
+    )
+
+    if args.one:
+        if not servers:
+            raise Exception('No servers')
+        if len(servers) > 1:
+            raise Exception('Multiple servers')
+
+    changes = bool(args.reset or args.update)
+    for server in servers:
+        print(server)
+        if changes:
+            apply_changes(server, args.reset, args.update)
+        print_server(server, args.attrs)
+    if changes:
+        servers.commit()
```

Comment:
Will it print errors?

Reply:
It wouldn't `print` anything, but raise some kind of `Exception`.

Reply:
Yeah, that's fair enough I guess. Or do you also want to notify the user if the changes actually did work at the end?

Reply:
I take the Unix approach of not emitting anything when everything is fine.

Reply:
Alright then. 👍